src/rgw/rgw_gc.cc

```diff
@@ -418,14 +579,14 @@ int RGWGC::process(int index, int max_secs, bool expired_only,
       if (now >= end) {
         goto done;
       }
-#if 0
-      if (chain.objs.empty()) {
-        io_manager.schedule_tag_removal(index, info.tag);
-      } else {
-        io_manager.add_tag_io_size(index, info.tag, chain.objs.size());
+      if (! transitioned_objects_cache[index]) {
+        if (chain.objs.empty()) {
+          io_manager.schedule_tag_removal(index, info.tag);
+        } else {
+          io_manager.add_tag_io_size(index, info.tag, chain.objs.size());
+        }
       }
-#endif
-  if (! chain.objs.empty()) {
+      if (! chain.objs.empty()) {
```

Comment:
@cbodley : In case rgw_init_ioctx() returns failure, should we break out of the loop for queue entries, instead of processing the next one and try next time?